src/server/scripts/EasternKingdoms/Karazhan/boss_midnight.cpp

```diff
@@ -151,14 +151,8 @@ struct boss_attumen : public BossAI
         {
             if (Creature* midnight = instance->GetCreature(DATA_MIDNIGHT))
             {
-                if (midnight->GetHealth() > me->GetHealth())
-                {
-                    summon->SetHealth(midnight->GetHealth());
-                }
-                else
-                {
-                    summon->SetHealth(me->GetHealth());
-                }
+                summon->SetHealth(std::max(midnight->GetHealth(), me->GetHealth()));
+                summon->SetMaxHealth(summon->GetHealth());
```

Comment:
I dont think this is correct, max health never changed just the current pct is

Reply:
The SetHealth() value is unchanged, only refactored

Issue is he spawns at <100% hp, so to match source the max health is set to current health 

https://www.youtube.com/watch?v=kul6efg1hZA&t=150s

![image](https://github.com/user-attachments/assets/d67cb8db-dc0f-466f-bfbf-e6e7297de737)

Reply:
Observed in a 2.5.4 43861 sniff
```
Packet number 123048:
Creature 15550 has 345495 health.
Creature 16151 has 95611 health.

Packet number 123057:
Creature 16152 is created, has both max health and current health of 379400.
```
The few sniffs I have don't have this guy spawning with any weird health values, *but* shortly after spawn (~3.3s after in the above packet's case), it looks like his current (not max) health drops near to the higher value, i.e. 342760 (likely includes the damage 16152 already took).

Reply:
So max health isnt changed